Wobble.Tests/Screens/Tests/TaskHandler/TaskHandlerScreenView.cs

```diff
@@ -3,41 +3,154 @@
 using System.Text;
 using Microsoft.Xna.Framework;
 using Quaver.Shared.Scheduling;
+using Wobble.Assets;
+using Wobble.Graphics;
+using Wobble.Graphics.Sprites;
+using Wobble.Graphics.UI.Buttons;
 using Wobble.Screens;
 
 namespace Wobble.Tests.Screens.Tests.TaskHandler
 {
     public class TaskHandlerScreenView : ScreenView
     {
+        /// <summary>
+        ///    Task Handler for testing
+        /// </summary>
         public TaskHandler<int, int> RandomNumbersTask { get; }
 
+        /// <summary>
+        ///    Random Number Generator
+        /// </summary>
+        private Random RNG { get; } = new Random();
 
+        /// <summary>
+        ///    User will press this to Run Random Numbers Task
+        /// </summary>
+        private TextButton TestButton { get; }
 
+        /// <summary>
+        ///    Displays progression of Random Numbers Task
+        /// </summary>
+        private SpriteText ProgressionText { get; }
+
+        /// <summary>
+        ///    Displays result of Random Numbers Task
+        /// </summary>
+        private SpriteText ResultText { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="screen"></param>
         public TaskHandlerScreenView(Screen screen) : base(screen)
         {
-            RandomNumbersTask = new TaskHandler<int, int>();
+            RandomNumbersTask = new TaskHandler<int, int>(TestFunction);
             RandomNumbersTask.OnCompleted += OnTaskComplete;
+            RandomNumbersTask.OnCancelled += OnTaskCancelled;
+
+            TestButton = new TextButton(WobbleAssets.WhiteBox, "exo2-medium", "Compute Random Number", 16)
+            {
+                Parent = Container,
+                Alignment = Alignment.MidCenter,
+                Size = new ScalableVector2(350, 60),
+                Text =
+                {
+                    Tint = Color.Black,
+                },
+                Y = -60
+            };
+            TestButton.Clicked += ButtonPressed;
+
+            ProgressionText = new SpriteText("exo2-medium", "", 14)
+            {
+                Parent = Container,
+                Alignment = Alignment.MidCenter,
+                Size = new ScalableVector2(250, 60),
+                Tint = Color.Black,
+                Y = 10
+            };
+
+            ResultText = new SpriteText("exo2-medium", "", 14)
+            {
+                Parent = Container,
+                Alignment = Alignment.MidCenter,
+                Size = new ScalableVector2(250, 60),
+                Tint = Color.Black,
+                Y = 40
+            };
         }
 
-        public void OnTaskComplete(object sender, TaskCompleteEventArgs<int, int> args)
+        /// <summary>
+        ///    Call when the button is pressed.
+        /// </summary>
+        private void ButtonPressed(object sender, EventArgs args)
```

Comment:
Forget the last comment. This part specifically is unclear because it starts a new task every time you click it. Instead I'd expect the button to act as a `Start` and `Stop` mechanism.

Reply:
Ok, I've updated the screen. The button will now act as a toggle.